flutter_app/test/widget/display_all_logs_widget_test.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:imacs/modules/get_airside_logs.dart';
+import 'package:imacs/screens/log_displayer_screen.dart';
+import 'package:imacs/widgets/display_all_logs_widget.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group(
+    'Display All Logs Widget',
+    () {
+      testWidgets(
+        'Displays airside logs by showing their respective path',
+        (WidgetTester tester) async {
+          const String pathToDirectory = 'C:\\Users\\emmao\\Documents\\logs';
+          final getAirsideLogs =
+              GetAirsideLogs(pathToDirectory: pathToDirectory);
+
+          await tester.pumpWidget(
+            MaterialApp(
+              home: Scaffold(
+                body: LogsList(
+                  getAirsideLogs:
+                      GetAirsideLogs(pathToDirectory: pathToDirectory),
+                ),
+              ),
+            ),
+          );
+          await tester.pump();
+          final expectedFiles = getAirsideLogs.getFiles();
+
+          final lastFile = expectedFiles.last;
+          final lastFileSegments = lastFile.uri.pathSegments;
+          final lastFileName = lastFileSegments.length == 1
+              ? lastFileSegments.last
+              : lastFileSegments
+                  .sublist(lastFileSegments.length - 2, lastFileSegments.length)
+                  .join('/');
+
+          final firstFile = expectedFiles.first;
+          final firstFileSegments = firstFile.uri.pathSegments;
+          final firstFileName = firstFileSegments.length == 1
+              ? firstFileSegments.last
+              : firstFileSegments
+                  .sublist(
+                      firstFileSegments.length - 2, firstFileSegments.length)
+                  .join('/');
+
+          //testing scroll
+          await tester.drag(
+              find.byType(SingleChildScrollView), const Offset(0, -400));
+          await tester.pumpAndSettle();
+
+          expect(find.text(lastFileName),
+              findsOneWidget); // don't know if this effectively checks if it can scroll
```

Comment:
Scrolling isn't something that can be tested well, checking for a `ScollView` or `ListView` widget usually satisfies though. Usually `tester.drag` is used for stuff like scrolling to a new page or initiating a function or something, but what you have done here is still a valid test.